plugins/importtext_procedure.cc

```diff
@@ -0,0 +1,1154 @@
+#include "mldb/server/dataset_context.h"
+#include "mldb/vfs/filter_streams.h"
+#include "mldb/vfs/fs_utils.h"
+
+
+using namespace std;
+
+namespace Datacratic {
+namespace MLDB {
+
+static constexpr size_t ROWS_PER_CHUNK=65536;
+
+DEFINE_STRUCTURE_DESCRIPTION(ImportTextConfig);
+
+ImportTextConfigDescription::ImportTextConfigDescription()
+{
+    addField("dataFileUrl", &ImportTextConfig::dataFileUrl,
+             "URL of the text data to import");
+    addField("ouputDataset", &ImportTextConfig::ouputDataset,
+             "Dataset to record the data into.",
+             PolyConfigT<Dataset>().withType("tabular"));
+    addField("headers", &ImportTextConfig::headers,
+             "List of headers for when first row doesn't contain headers",
+             vector<Utf8String>());
+    addField("quotechar", &ImportTextConfig::quoter,
+             "Character to enclose strings", string("\""));
+    addField("delimiter", &ImportTextConfig::delimiter,
+             "Delimiter for column separation", string(","));
+    addField("limit", &ImportTextConfig::limit,
+             "Maximum number of lines to process.  Bad lines including empty lines "
+             "contribute to the limit.  As a result, it is possible for the dataset "
+             "to contain less rows that the requested limit.");
+    addField("offset", &ImportTextConfig::offset,
+            "Skip the first n lines (excluding the header if present).", int64_t(0));
+    addField("encoding", &ImportTextConfig::encoding,
+             "Character encoding of file: 'us-ascii', 'ascii', 'latin1', 'iso8859-1', 'utf8' or 'utf-8'",
+             string("utf-8"));
+    addField("ignoreBadLines", &ImportTextConfig::ignoreBadLines,
+             "If true, any line causing a parsing error will be skipped. "
+             "Empty lines are considered bad lines.", false);
+    addField("replaceInvalidCharactersWith",
+             &ImportTextConfig::replaceInvalidCharactersWith,
+             "If this is set, it should be a single Unicode character will be used "
+             "to replace badly-encoded characters in the input. "
+             "The default is nothing, which will cause lines with badly-"
+             "encoded characters to throw an error.");
+    addField("select", &ImportTextConfig::select,
+             "Which columns to use.",
+             SelectExpression::STAR);
+    addField("where", &ImportTextConfig::where,
+             "Which lines to use to create rows.",
+             SqlExpression::TRUE);
+    addField("named", &ImportTextConfig::named,
+             "Row name expression for output dataset. Note that each row "
+             "must have a unique name.",
+             SqlExpression::parse("lineNumber()"));
+    addField("timestamp", &ImportTextConfig::timestamp,
+             "Expression for row timestamp.",
+             SqlExpression::parse("fileTimestamp()"));
+
+    addParent<ProcedureConfig>();    
+    onUnknownField = [] (ImportTextConfig * config,
```

Comment:
This can probably be removed now.  It was added to help with migration, but by definition everything that was using this is being migrated.